src/fs/rock/RockSwapDir.cc

```diff
@@ -867,6 +891,17 @@ Rock::SwapDir::writeCompleted(int errflag, size_t, RefCount< ::WriteRequest> r)
         CollapsedForwarding::Broadcast(*sio.e);
 }
 
+void
+Rock::SwapDir::remoteIndexingCompleted()
+{
+    assert(IamWorkerProcess());
+
+    if(!theFile || theFile->error())
+        return; // the Rock db file is not open/initialized
```

Comment:
Please double check our logic here. I am concerned that we are not calling storeRebuildComplete() on these errors, even though the indexing has stopped. Without that call, `squid -F` will probably get stuck, and even a "regular" Squid instance may experience problems while waiting (forever) for indexing to finish.